packages/nextjs/src/common/wrapPageComponentWithSentry.ts

```diff
@@ -0,0 +1,66 @@
+import { captureException } from '@sentry/core';
+import { addExceptionMechanism } from '@sentry/utils';
+
+interface FunctionComponent {
+  (...args: unknown[]): unknown;
+}
+
+interface ClassComponent {
+  new (...args: unknown[]): {
+    render(...args: unknown[]): unknown;
+  };
+}
+
+function isReactClassComponent(target: unknown): target is ClassComponent {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  return typeof target === 'function' && target?.prototype?.isReactComponent;
+}
+
+/**
+ * Wraps a page component with Sentry error instrumentation.
+ */
+export function wrapPageComponentWithSentry(pageComponent: FunctionComponent | ClassComponent): unknown {
+  if (isReactClassComponent(pageComponent)) {
+    return class SentryWrappedPageComponent extends pageComponent {
+      public render(...args: unknown[]): unknown {
```

Comment:
l: I wonder if we can use suspense + error boundary to catch this - it's what is recommended with https://github.com/reactjs/rfcs/pull/215

Reply:
SSR error boundaries are effectively only needed when you want to display a fallback in case of an error. I don't think we need this and I don't think this is possible in non React Server Component worlds but there's an excellent article on this: https://gal.hagever.com/posts/react-error-boundaries-and-ssr